contrib/devtools/README.md

```diff
@@ -2,6 +2,47 @@ Contents
 ========
 This directory contains tools for developers working on this repository.
 
+archive-binaries.py
+===============
+
+Script to copy all the binaries (bitcoind, bitcoin-tx, bitcoin-cli,
+qt/bitcoin-qt) to a bin and bin/qt directories (assumed to exist in
+the calling directory), appending the last commit id (and optionally
+an arbitrary string before the commit id passed ar parameter before
+the commit id) to the file name. 
+
+This way one can keep a collection of binaries for many different
+branches for testing (while clearly ditinguishing which is which).
+
+Example usage:
+
+```
+git fetch origin
+git checkout origin/v0.12.0
+make check -j21
+python contrib/devtools/archive-binaries.py v0.12.0
+./bin/bitcoind-v0.12.0-188ca9c
+
+git checkout origin/master
+git clean -fdx -e bin/
```

Comment:
I wonder if we could start maintaining a file for those to later git clean -e cat src/.git_clean_exclude_files. It wouldn't need to be complete for older versions, just start now and if people find time to find more stuff to include for the older versions, you can add that too.
Oh, sorry, actually what we want is to explicitly include files for cleaning, not exclude them. Is there an option for that?
To be clear, the use case for this cleanup is time travelling to very old versions and coming back, in case anybody else wants to try what cleaning problems you may encounter.
Maybe my way of solving them is not the right one, I just needed to do that at some point, for something related to libsecp if I remember correctly.
IIRC, it was @laanwj or @theuni who told me about ```git clean -fdx```, but could have been also anybody else in #bitcoin-dev .
I can guarantee I didn't discovered -fdx by myself.